### PR TITLE
changed json deserialization error to 400 from 500 on POST

### PR DESCRIPTION
**Pull request by shawnrusaw-wf**
@aaronkavlie-wf @tylertreat 

fixed POST and PUT so they 400 on invalid json, rather than 500

### CODE REVIEW

**Comment by rmconsole-wf** [comment]
## SOC 1 Checklist
- [ ] All commits reviewed (7274431fd92ca8d05bfc7c18adb5a5722c84337d)
- [ ] QA Review
- [x] Security Review Completed if Applicable:  (N/A)

To remain in SOC1 Compliance the following users cannot merge this PR:
* shawnrusaw-wf

***
*Last updated on Wednesday, April 29 12:40 PM CST*

**Comment by tylertreat-wf** [comment]
+1

**Comment by aaronkavlie-wf** [comment]
+1

Will this still do the right thing without the explicit `ctx.setStatus()` call?

**Comment by shawnrusaw-wf** [comment]
i'm still trying to figure out how to test that. this is what PUT does, so i'm guessing yes

**Comment by tylertreat-wf** [comment]
Yes, the response status will be pulled from the error if possible: https://github.com/Workiva/go-rest/blob/master/rest/serialize.go#L70

File: rest/handler.go [hunk 1]
```diff
@@ -138,8 +138,7 @@ func (h requestHandler) handleCreate(handler ResourceHandler) http.HandlerFunc {
 		data, err := decodePayload(payloadString(r.Body))
 		if err != nil {
 			// Payload decoding failed.
-			ctx = ctx.setError(err)
-			ctx = ctx.setStatus(http.StatusInternalServerError)
+			ctx = ctx.setError(BadRequest(err.Error()))
```
Comment (shawnrusaw-wf) [inline rest/handler.go]: POST

File: rest/handler.go [hunk 2]
```diff
@@ -274,8 +273,7 @@ func (h requestHandler) handleUpdate(handler ResourceHandler) http.HandlerFunc {
...
```
Comment (shawnrusaw-wf) [inline rest/handler.go]: PUT

**Comment by tylertreat-wf** [comment]
+1

**Comment by aaronkavlie-wf** [comment]
+1 to 2nd commit too :-)

**Comment by travisreed-wf** [comment]
rosie

**Pull request merged**